src/toxcore/dht/packet/ping_request.rs

```diff
@@ -150,12 +198,31 @@ impl ToBytes for PingRequestPayload {
     }
 }
 
+impl PingRequestPayload {
+    // Root parser for PingRequest object
+    fn parse<'a, E: ParseError<&'a [u8]>>(i: &'a [u8]) ->IResult<&'a [u8], PingRequestPayload, E> {
```

Comment:
Why is it a separate function? Why not just have this parsing inside `from_bytes5`?

Reply:
done